src/main/java/picard/arrays/ArraysCallingMetricAccumulator.java

```diff
@@ -0,0 +1,222 @@
+            if (genotype.isHet()) {
+                ++metric.numHets;
+            } else if (genotype.isHomVar()) {
+                ++metric.numHomVar;
+            }
+        }
+    }
+
+
+    /**
+     * Amends the provided metric with the data in the provided variant.  Also amends the summary metric re: reference bias.
+     */
+    private void updateSummaryMetric(final CollectArraysVariantCallingMetrics.ArraysVariantCallingSummaryMetrics metric,
+                                     final VariantContext vc,
+                                     final boolean hasSingletonSample) {
+        if (!vc.getCommonInfo().getFilters().contains(InfiniumVcfFields.ZEROED_OUT_ASSAY)) {
+            // A "zeroed-out SNP".  Marked as unusable/uncallable
+            metric.TOTAL_ASSAYS++;
+
+            metric.NUM_NO_CALLS += vc.getNoCallCount();
+            metric.NUM_CALLS += vc.getHomVarCount() + vc.getHetCount() + vc.getHomRefCount();
+            metric.NUM_AUTOCALL_CALLS += vc.getGenotypes().stream()
+                    .map(gt -> gt.getExtendedAttribute(InfiniumVcfFields.GTA, gt.getGenotypeString()))
+                    .filter(gta -> !gta.equals(VCFConstants.EMPTY_GENOTYPE))
+                    .count();
+
+            if (hasSingletonSample) {
+                ++metric.NUM_SINGLETONS;
+            }
+
+            if (vc.isSNP()) {
+                // Biallelic SNPs
+                final boolean isInDbSnp = dbsnp.snps.isDbSnpSite(vc.getContig(), vc.getStart());
+
+                metric.TOTAL_SNPS++;
+
+                if (isInDbSnp) {
+                    metric.NUM_IN_DB_SNP++;
+                }
+            } else if (vc.isIndel()) {
+                metric.TOTAL_INDELS++;
+            }
+        }
+
+        // Tally up the filtered SNPs & indels, then exit. The other metrics shouldn't be
+        // computed on low-confidence calls.
+        if (vc.isFiltered()) {
+            metric.FILTERED_SNPS++;
+        }
+    }
+
+    @Override
+    public Result result() {
+        final Collection<CollectArraysVariantCallingMetrics.ArraysVariantCallingDetailMetrics> values = sampleMetricsMap.values();
+        values.forEach(CollectArraysVariantCallingMetrics.ArraysVariantCallingDetailMetrics::updateDerivedValuesInPlace);
+
+        summaryMetric.updateDerivedValuesInPlace();
+        return new Result(summaryMetric, values);
+    }
+
+    public static class Result {
+        final CollectArraysVariantCallingMetrics.ArraysVariantCallingSummaryMetrics summary;
+        final Collection<CollectArraysVariantCallingMetrics.ArraysVariantCallingDetailMetrics> details;
+
+        Result(CollectArraysVariantCallingMetrics.ArraysVariantCallingSummaryMetrics summary,
+                      Collection<CollectArraysVariantCallingMetrics.ArraysVariantCallingDetailMetrics> details) {
+            this.summary = summary;
+            this.details = details;
+        }
+
+        public static Result merge(final Collection<Result> results) {
```

Comment:
I'm not sure I understand what this function is for, can we add a comment? It looks like it might be used to merge metrics if they were processed on different machines ?

Reply:
The tool / accumulator is designed to be multi-threaded.  This merge is to combine the results generated by each thread.

Reply:
I've added a comment.